### PR TITLE
New bot features

### PR DESCRIPTION
**Pull request by jimmyhmiller**
The diff is going to look crazy because I moved things to src.

I'm not sure how clear the idea behind the code will be. But if you checkout the allocate function, it is pretty straight forward. The goal is to be able to write all commands in as straight forward a way as possible. 

After this I'm going to have some bulk edits just modernizing things.

Feel free to ask any questions.

### CODE REVIEW

File: nomic-bot/src/universe/actions.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const action = (type, f=identity) => {
```
Comment (jimmyhmiller) [inline nomic-bot/src/universe/actions.js]: Need to fix this. Will make a PR soon.

**Comment by ArchmageInc** [comment]
Wow, okay, so only one question: Does this break anything @jimmyhmiller?

**Comment by dilbertKocik** [comment]
lol, yeah I was wondering the same thing. It looks pretty solid. But it's a lot of code to read through so who knows.

**Comment by jimmyhmiller** [comment]
I didn't test everything exhaustively, but I think it should work. If things do break, I'm happy to fix them. I've got a lot more planned for this, hopefully we can start having new additions become easier and easier.

**Comment by jimmyhmiller** [comment]
@ArchmageInc @dilbertKocik Any objections to me merging it?

**Comment by jimmyhmiller** [comment]
I actually did find a minor bug but have fixed it.

**Comment by ArchmageInc** [comment]
This is now built and pushed as archmageinc/nomic-bot:latest

**Comment by ArchmageInc** [comment]
archmageinc/nomic-bot:latest has been deployed. The previous version was archmageinc/nomic-bot:sawmill.

**Pull request merged**